pypdf/_writer.py

```diff
@@ -1808,85 +1808,101 @@ def removeLinks(self) -> None:  # deprecated
         deprecation_with_replacement("removeLinks", "remove_links", "3.0.0")
         return self.remove_links()
 
+    def remove_img_or_text(
```

Comment:
I would rather make this a private method:

```suggestion
    def _remove_img_or_text(
```

(that needs some adjustments in other parts as well)

Reply:
I would prefer to leave it public : currently `remove_images()` and `remove_text()` are cleaning the full document : this will provide a solution to just clean one page.
However I agree the name is too tricky : I propose to rename the function into `remove_image_or_text_from_page()`

Your opinion ?

Reply:
> ```
>     def remove_objects_from_page(
>         self, page: Union[PageObject, DictionaryObject], to_delete: ObjectDeletionFlag
>     ) -> None:
> ```

Sounds good to me.

> I see some advantages in the proposed interface:
>     1. We could add the capability to delete annotations (maybe even specific annotation types) in the same method

Text and Images requires to remove to parse content, that's why they are specific, however from a interface point of view we can merge them.

>     2. We could allow users to specify multiple flags at one method 

good idea implemented.

>     3. Personal preference: When I see an `_or_` in a method name, I tend to think that those should be two methods 😄
> 
     not iaw my rules...😉🤣🤣🤣🤣 
>     4. Personal preference: I think that the current use of `del_image=False` => delete text is not intuitive. I would like to avoid that.

btw I hate the name of the function `remove_links()` which name is not describing it properly. Shouldn't we have `remove_annots()` and `remove_links()` - for links only?

Reply:
I had to look it up - https://pypdf.readthedocs.io/en/latest/modules/PdfWriter.html#pypdf.PdfWriter.remove_links

Yes, that is exactly this kind of unexpected dual-purpose which I don't like -.-

I want to keep the amount of breaking changes low for this year. Give people a chance to adapt. But we should make a list of such things to be cleaned-up in future.

Additionally, the proposed remove_objects_from_page could support links and annotations as well (just by adding it to the `ObjectDeletionFlag` / letting users provide that value.